### DISCUSSION TITLE
Custom filteration in Stencil not working

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2555] [Q&A] [answered]
### Introduction

I'm trying the filter the elements in stencil based on some custom properties as follows:
![image](https://github.com/clientIO/joint/assets/124018900/93d6ef8d-28a6-4b42-ba19-f0bf19c5a00f)
Notice the expected values in the console:
![image](https://github.com/clientIO/joint/assets/124018900/2e6e7922-2d7d-4012-ba02-39244063a18f)
This works and filters as expected. However, when I add an additional field based on which I want filtering to happen, the logic fails.
![image](https://github.com/clientIO/joint/assets/124018900/f3436718-2c13-4bf1-ba31-5e07ac4d24a3)
![image](https://github.com/clientIO/joint/assets/124018900/9207ba2b-d045-4dd6-b3b3-c5e07dbc83aa)
I understand that the culprit is the 'element' object, that doesn't have valid properties. This issue happens when I add the additional OR condition in the return statement. 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
You are trying to call `includes()` on `undefined`.
You need to check if the property is a string first (or fallback to an empty string when the property does not exist).
